### DISCUSSION TITLE
Add option to cache node names across transformers

### DISCUSSION THREAD

**Discussion by stanhu** [rgrove/sanitize#217] [Ideas]
In 4.6.2, https://github.com/rgrove/sanitize/commit/caa558a25886638427d3005030260af8518612ea significantly cut down on the memory usage by downcasing the node name once.

However, someone reported in https://github.com/rgrove/sanitize/issues/177 reported that this broke certain transformers that modified the node name since the name was cached.

I'd suggest that it would still be nice to have a way to support that optimization for applications that don't change the node name. In my sample test, I see that 11 MB of RAM is being allocated, and with thousands of these documents being processed every second on GitLab, we are seeing slowdowns due to garbage collection. 

Thoughts?

**Comment by rgrove** [comment]
Seems like a simple solution might be to reimplement the change from caa558a but compare `node_name` with `node.name` after each iteration and update it if the node name changes.
